File: samples/SampleCounterWithDomView/Views/Renderer.cs
using Metano.TypeScript.DOM;

namespace SampleCounterWithDomView.Views;

public sealed class Renderer
{
    private readonly HtmlElement _container;

    public Renderer(string containerId)
    {
        _container = Js.Document.GetElementById(containerId) ?? CreateElement(containerId);
    }

    private HtmlElement CreateElement(string containerId)
    {
        var element = Js.Document.CreateElement<HtmlDivElement>();
        element.Id = containerId;
        Js.Document.Body.Append(element);

        return element;
    }

    public void Render(IView view) { }
}

File: targets/js/sample-counter-v3/lib/views/renderer.js
export class Renderer {
    _container;
    constructor(containerId) {
        this._container = document.getElementById(containerId) ?? this.createElement(containerId);
    }
    createElement(containerId) {
        const element = document.createElement("div");
        element.id = containerId;
        document.body.append(element);
        return element;
    }
    render(view) {
        view.render(this._container);
    }
}
